client/test/progression.go

```diff
@@ -104,16 +106,17 @@ func (r *Paul) exec(_cfg ExecConfig, ch *paymentChannel) {
 		s.Balances[assetIdx][1] = big.NewInt(half)
 	}))
 
-	// await our progression confirmation
-	<-r.progressed
-	r.t.Logf("%v received progression confirmation 1", r.setup.Name)
-
-	// await them progressing
-	progEvent := <-r.progressed
-	r.t.Logf("%v received progression confirmation 2", r.setup.Name)
+	// Await progressed event 1.
+	r.log.Debugf("%v awaiting progressed event 1", r.setup.Name)
+	e := <-r.progressed
+	assert.Truef(e.Version() == 1, "expected version 1, got version %v", e.Version())
+	r.waitStage()
```

Comment:
What does this `waitStage` do exactly?
It waits until a new stage has been started / the current stage has been completed s.t. we are expected to get a new progressed event?

Reply:
It's a sync point between both test routines I afaik.

Reply:
Yes, it's an artificial synchronization point that we use in the tests. Ideally, I think, we should be able to make all tests work without those, but in some cases this would potentially require a rework of some architecture parts.